internal/declarative/v2/reconciler.go

```diff
@@ -592,6 +600,10 @@ func (r *Reconciler) getTargetClient(ctx context.Context, obj Object) (Client, e
 			return nil, fmt.Errorf("failed to patch namespace: %w", err)
 		}
 	}
+	err = testClient(ctx, clnt)
```

Comment:
If you testClient here, it means for all client connection, you add additionally a request, which is not necessary, because anyway, if the connection have problems, the process afterwards which doing real meaningful requests, like deploy resources, get module data will be failed and return back error, you don't have to test this connection upfront, besides, even if it pass the connection test in this step, can you make sure the followup process not failed with connect problem? With this additional tests, however it brings some side effects to affect the performance, e.g: slow down the reconciliation time, add networking costs.

Reply:
That is on purpose to test the connection upfront. because in case of orphaned kyma, the downstream process generates a huge number og logs before the reconciliation is finished. that is the problem that this PR is trying to solve